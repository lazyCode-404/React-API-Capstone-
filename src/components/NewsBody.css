@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300&display=swap');

body {
  width: 100%;
  margin: auto;
  font-family:
    -apple-system,
    BlinkMacSystemFont,
    'Segoe UI',
    'Roboto',
    'Oxygen',
    'windows',
    'Cantarell',
    'Fira Sans',
    'Droid Sans',
    'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family:
    source-code-pro,
    Menlo,
    Monaco,
    Consolas,
    'Courier New',
    monospace;
}

.Search-sec {
  background-color: #cd4275;
  display: flex;
  flex-direction: column;
  width: inherit;
}

.container-fluid {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-weight: 700;
  background: #cd4275;
  color: #fff;
  list-style: none;
}

button {
  display: flex;
  justify-content: center;
  align-items: center;
  background: #fff;
  border: none;
  padding: 5px;
}

.selected-name {
  text-align: center;
  font-size: 30px;
}

.today {
  list-style: none;
  font-size: 18px;
  text-align: center;
}

.total {
  text-align: center;
  font-size: 16px;
  list-style: none;
}

.list-country {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  justify-content: center;
  align-items: center;
  gap: 5px;
  width: inherit;
  list-style: none;
  font-weight: 600;
}

.list-country a:nth-child(4n),
.list-country a:nth-child(4n + 1) {
  background-color: #ec4e8a;
  color: #fff;
}

.list-details {
  display: flex;
  flex-direction: column;
  color: #fff;
  gap: 39px;
}

.list-details:hover {
  background: radial-gradient(black, transparent);
  width: 95%;
}

.a {
  text-decoration: none;
}

img {
  width: 155px;
}

.header {
  display: flex;
  justify-content: space-around;
  align-items: center;
  background: #ec4e8a;
  color: #fff;
  font-weight: 600;
  font-weight: 700;
  box-shadow: 4px 4px 4px 4px;
}

.header-actions {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 12px;
}

.form-input {
  outline: none;
  padding: 3px;
  border: none;
  border-radius: 5px;
  width: 100%;
}

.area {
  text-align: center;
  font-size: 20px;
  color: #fff;
  font-weight: 600;
}

.form {
  display: flex;
  justify-content: center;
  align-items: center;
}

.search-bar {
  display: flex;
  gap: 3px;
  justify-content: center;
  align-items: center;
}

.name {
  font-size: 20px;
}

.population {
  margin-top: -19px;
  font-size: 20px;
}

@media (min-width: 768px) {
  .list-country {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    margin: auto 0;
    gap: 5px;
    width: 95%;
  }

  .list-details {
    padding: 12px;
    width: 80%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }

  .name {
    font-size: 20px;
    margin-top: -54px;
  }

  .form {
    margin-bottom: 30px;
  }

  .form-input {
    outline: none;
    padding: 5px;
    border: none;
    border-radius: 5px;
    width: 253px;
  }
}
