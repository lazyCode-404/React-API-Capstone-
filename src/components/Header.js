import React from 'react';
import { useNavigate } from 'react-router-dom';
import { BsMicFill, BsFillGearFill, BsCaretLeftFill } from 'react-icons/bs';
import './NewsBody.css';

const Header = () => {
  const navigate = useNavigate();

  return (
    <nav className="header">
      {
        window.location.pathname === '/'
          ? (
            <span>Home</span>
          )
          : (
            <button type="button" onClick={() => navigate(-1)}>
              <BsCaretLeftFill />
              Back
            </button>
          )
      }
      <h1>COVID-19 Data Statistics</h1>
      <div className="header-actions">
        <span><BsMicFill /></span>
        <span><BsFillGearFill /></span>
      </div>
    </nav>
  );
};

export default Header;
