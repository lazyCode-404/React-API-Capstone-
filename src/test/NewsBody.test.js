import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import store from '../redux/configureStore';
import NewsBody from '../components/NewsBody';

describe('Render components', () => {
  test('render navbar', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Header /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render homePage', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><NewsBody /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
