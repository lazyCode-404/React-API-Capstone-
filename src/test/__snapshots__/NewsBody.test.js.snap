// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render homePage 1`] = `
<div
  className="Search-sec"
>
  <h3
    className="area"
  >
    Africa
  </h3>
  <form
    className="form"
  >
    <div
      className="search-bar"
    >
      <input
        className="form-input"
        onChange={[Function]}
        placeholder="Search country"
        type="text"
        value=""
      />
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </div>
  </form>
  <ul
    className="list-country"
  />
</div>
`;

exports[`Render components render navbar 1`] = `
<nav
  className="header"
>
  <span>
    Home
  </span>
  <h1>
    COVID-19 Data Statistics
  </h1>
  <div
    className="header-actions"
  >
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
    </span>
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
        />
      </svg>
    </span>
  </div>
</nav>
`;
